crypto/evp/pkey_mac.c

```diff
@@ -276,8 +318,18 @@ static int pkey_mac_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
         break;
 
     case EVP_PKEY_CTRL_SET_DIGEST_SIZE:
-        return EVP_MAC_ctrl(hctx->ctx, EVP_MAC_CTRL_SET_SIZE, (size_t)p1);
+        {
+            OSSL_PARAM params[2];
+            size_t params_n = 0;
+            size_t size = (size_t)p1;
 
+            params[params_n++] =
+                OSSL_PARAM_construct_size_t("outlen", &size);
+            params[params_n] = OSSL_PARAM_construct_end();
+
+            return EVP_MAC_CTX_set_params(hctx->ctx, params);
```

Comment:
Should we check if it was successfully set?

Reply:
Hah, that discussion does proliferate into all kinds of places, isn't it?  Yeah, I'll add a check similar to #9562, with a TODO comment to change it when we have a better working mechanism.

Reply:
Done